### PR TITLE
Geometry_Engine: redundant calls to clone removed

### LINKED ISSUE #943: Geometry_Engine: Remove unnecessary clones
e.g. in
```
        public static BoundingBox Bounds(this BoundingBox boundingBox)
        {
            return boundingBox.Clone();
        }
```

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #943

### Test files

@FraserGreenroyd as discussed, it would be great if you could do a sanity check and onboard someone to dust off the test scripts that would confirm that I have not broken anything 🙈

### CODE REVIEW

**Comment by Tom-Kingstone** [comment]
@BHoMBot check unit-tests

File: Geometry_Engine/Query/IsCoplanar.cs [hunk 1]
```diff
@@ -88,8 +88,8 @@ public static bool IsCoplanar(this Line line1, Line line2, double tolerance = To
 
         public static bool IsCoplanar(this Polyline curve1, Polyline curve2, double tolerance = Tolerance.Distance)
         {
-            List<Point> cPts = curve1.DeepClone().ControlPoints;
-            cPts.AddRange(curve2.DeepClone().ControlPoints);
+            List<Point> cPts = curve1.ControlPoints;
+            cPts.AddRange(curve2.ControlPoints);
```
[the remark this answers is not in the document]
Comment (pawelbaran) [reply Geometry_Engine/Query/IsCoplanar.cs:92]: Ah yes, of course - looks like I made a really dumb mistake 🙈 Pushing the fix now 👍

**Comment by pawelbaran** [comment]
Thanks @Tom-Kingstone for testing, I pushed a fix for the bug that you have found 👍

**Comment by Tom-Kingstone** [comment]
@BHoMBot check unit-tests

**Review by Tom-Kingstone — APPROVED** [review]
After testing with Geometry_Engine unit tests locally, as well as some other engines that are dependent on the Geometry_Engine (namely environment, facade and physical), I am happy to say that removing DeepClone in these places does not break the methods.

**Review by FraserGreenroyd — APPROVED** [review]
Have reviewed the code and @Tom-Kingstone has run the tests so this looks good to me now 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer
@BHoMBot check compliance
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: core, documentation-compliance, null-handling, serialisation, unit-tests

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**